lib/elixir/lib/code/formatter.ex

```diff
@@ -1487,16 +1487,29 @@ defmodule Code.Formatter do
     end
   end
 
-  defp atom_to_algebra(atom) when atom in [nil, true, false] do
+  defp atom_to_algebra(atom, _) when atom in [nil, true, false] do
     Atom.to_string(atom)
   end
 
   # TODO: Remove this clause in v1.16 when we no longer quote operator :..//
-  defp atom_to_algebra(:"..//") do
+  defp atom_to_algebra(:"..//", _) do
     string(":\"..//\"")
   end
 
-  defp atom_to_algebra(atom) do
+  defp atom_to_algebra(:\\, meta) do
+    string =
+      case Keyword.get(meta, :delimiter) do
+        "\"" ->
+          ":\"\\\\\""
+
+        _ ->
+          ":\\\\"
+      end
```

Comment:
```suggestion
  defp atom_to_algebra(:\\, meta) do
    string =
      # Since we parse strings without unescaping, the atoms
      # :\\ and :"\\" have the same representation, so we need
      # to check the delimiter and handle them accordingly.
      case Keyword.get(meta, :delimiter) do
        "\"" -> ":\"\\\\\""
        _ -> ":\\\\"
      end
```